packages/client/src/editor.tsx

```diff
@@ -6,24 +6,30 @@ import CodeMirror from "codemirror";
 import "codemirror/mode/markdown/markdown.js";
 import "codemirror/lib/codemirror.css";
 import { WebsocketProvider } from "y-websocket";
+import { useSelector } from "react-redux";
+import { selectDocument } from "./redux/document";
 
-const ROOM_NAME = "new-room";
 const SIGNALLING_SERVICE =
   process.env.STORYBOOK_SIGNAL_URL || "ws://localhost:6006/ws/signal";
 const PROVIDER_SERVICE =
   process.env.STORYBOOK_PROVIDER_URL || "ws://localhost:6006/ws/provider";
 
 export default function Editor() {
   const ref = React.useRef();
+  const doc = useSelector(selectDocument);
+
   React.useEffect(() => {
+    if (!doc) {
+      return;
+    }
```

Comment:
adding this dependency, in an effect, is going to cause problems. It will be better if we extract this dependency outside the component entirely, and prevent the component from mounting if a document is not selected.

```tsx
function DocumentEditor({document_id}){
  const doc = useSelector(selectDocument);
  if(!doc){
    return <p>Select a document</p>
  }
  return <Editor document_id={document_id}/>
}

```